code/modules/projectiles/guns/energy.dm

```diff
@@ -497,6 +492,10 @@
 	var/ammo = null
 	///how long it takes between each shot of that mode, same as gun fire delay.
 	var/fire_delay = 0
+	///Mostly for the laser carbine.
+	var/burst_amount = 4
+	///gives firemode selections for guns.
+	var/gun_firemode_list = null
```

Comment:
this isnt typycast as a list and dont do = null in declarations since it's implied

Reply:
Standing request